City-of-Helsinki/employment-services-ui
# Thf 610 storage url filter

### Changes

- Change filter styling
- Added filter tag to the session storage
- Added locale to the session storage
- Added scroll from top position to the session storage

<img width="1021" alt="Screenshot 2023-10-11 at 12 32 48" src="https://github.com/City-of-Helsinki/employment-services-ui/assets/42113018/ec6b54ee-798d-4f0c-b900-6b32c844699a">

Some of the functions has been changed so that those are easy to use with different kind of tag.

#### Testing

##### Test 1
- GO http://localhost:3000/ajankohtaista/tapahtumat
- Choose Young people and immigrants you should get URL below
- http://localhost:3000/en/current-matters/events?tag=young%20people&tag=immigrants
- Copy the URL
- Open new tab and place the copied URL to the tab
- You should get page where young people and immigrants filters are choosen.


##### Test 2

- GO http://localhost:3000/ajankohtaista/tapahtumat
- Choose a filter
- Scroll down the screen little bit
- Choose an event and got to the page of the event. Stay the same tab. 
- Come back with breadcrumbs or with browser's back button. (try both)
- You should get back to same place on the page that you left. 
- If you scroll up you should also have the same filter that you had when you left and the page URL should also have the filter

##### Test 3

- GO http://localhost:3000/ajankohtaista/tapahtumat
- Select some filters
- Go to some other page using navigation. 
- Then come back to the event page using navigation
- You shouldn't have any filters and your scroll position should be normal.


##### Test 4
- GO http://localhost:3000/ajankohtaista/tapahtumat
- Select some filters
- Go to some other page using navigation. 
- Then come back to the event page using browser back button
- You should have the same filters selected and filters on the URL.

##### Test 5
- GO http://localhost:3000/ajankohtaista/tapahtumat
- Select some filters. 
- Open new tab and go to http://localhost:3000/ajankohtaista/tapahtumat
- Your shouldn't have any filters selected.
